pymc4/distributions/mixture.py

```diff
@@ -19,11 +26,36 @@ class Mixture(Distribution):
     ----------
     p : array of floats
         p >= 0 and p <= 1
-        the mixture weights, in the form of probabilities
+        the mixture weights, in the form of probabilities, must sum to one.
     distributions : multidimensional PyMC4 distribution (e.g. `pm.Poisson(...)`)
         or iterable of one-dimensional PyMC4 distributions the
         component distributions :math:`f_1, \ldots, f_n`
     """
 
     def __init__(self, name, p, distributions, **kwargs):
         super().__init__(name, p=p, distributions=distributions, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions, **kwargs):
+        p, d = conditions["p"], conditions["distributions"]
+        # if 'd' is a list of pymc distributions, then use the underlying
+        # tfp distributions for the mixture
+        if isinstance(d, list):
+            distributions = [el._distribution for el in d]
```

Comment:
In this branch, we should always test that every element of `d` is a `pm.Distribution`and raise a `TypeError` otherwise.